community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/MandatoryPropertyChecker.java

```diff
@@ -0,0 +1,312 @@
+
+    private static abstract class Checker<RECORD extends PrimitiveRecord,
+            REPORT extends ConsistencyReport.PrimitiveConsistencyReport>
+            implements OwningRecordCheck<RECORD,REPORT>
+    {
+        private final OwningRecordCheck<RECORD,REPORT> next;
+
+        private Checker( OwningRecordCheck<RECORD,REPORT> next )
+        {
+            this.next = next;
+        }
+
+        @Override
+        public void check( RECORD record, CheckerEngine<RECORD,REPORT> engine, RecordAccess records )
+        {
+            next.check( record, engine, records );
+            if ( record.inUse() )
+            {
+                PrimitiveIntSet keys = mandatoryPropertiesFor( record, engine, records );
+                if ( keys != null )
+                {
+                    checkProperties( record, keys, engine, records );
+                }
+            }
+        }
+
+        private void checkProperties( RECORD record, PrimitiveIntSet keys, CheckerEngine<RECORD,REPORT> engine,
+                RecordAccess records )
+        {
+            long firstProperty = record.getNextProp();
+            if ( !Record.NO_NEXT_PROPERTY.is( firstProperty ) )
+            {
+                engine.comparativeCheck( records.property( firstProperty ), new ChainCheck<RECORD,REPORT>( keys ) );
+            }
+            else
+            {
+                reportMissingKeys( engine.report(), keys );
+            }
+        }
+
+        abstract PrimitiveIntSet mandatoryPropertiesFor( RECORD record, CheckerEngine<RECORD,REPORT> engine,
+                RecordAccess records );
+
+
+        @Override
+        public ComparativeRecordChecker<RECORD,PrimitiveRecord,REPORT> ownerCheck()
+        {
+            return next.ownerCheck();
+        }
+
+        @Override
+        public void checkChange( RECORD oldRecord, RECORD newRecord, CheckerEngine<RECORD,REPORT> engine,
+                DiffRecordAccess records )
+        {
+            next.checkChange( oldRecord, newRecord, engine, records );
+        }
+    }
+
+    static class NodeChecker extends Checker<NodeRecord,ConsistencyReport.NodeConsistencyReport>
+    {
+        private final PrimitiveIntObjectMap<int[]> mandatory;
+        private final int capacity;
+
+        private NodeChecker( PrimitiveIntObjectMap<int[]> mandatory,
+                OwningRecordCheck<NodeRecord,ConsistencyReport.NodeConsistencyReport> next )
+        {
+            super( next );
+            this.mandatory = mandatory;
+            this.capacity = SizeCounter.countSizeOf( mandatory );
+        }
+
+        @Override
+        PrimitiveIntSet mandatoryPropertiesFor( NodeRecord record,
+                CheckerEngine<NodeRecord,ConsistencyReport.NodeConsistencyReport> engine, RecordAccess records )
+        {
+            PrimitiveIntSet keys = null;
+            for ( Long label : getListOfLabels( record, records, engine ) )
+            {
+                int[] labelKeys = mandatory.get( label.intValue() );
```

Comment:
I think this variable should be named `propertyKeys`. Same for one in `RelationshipChecker#mandatoryPropertiesFor`